### PR TITLE
Retry pulling images from docker for warm up

### PR DESCRIPTION
**Pull request by anero**
:canned_food: Retries 5 times with 5 sec waits in between.

### CODE REVIEW

File: packer/linux/conf/bin/bk-install-elastic-stack.sh [hunk 1]
```diff
@@ -239,8 +239,17 @@ systemctl start "buildkite-agent"
...
-# allow failures while warming the images we use.
-docker pull circleci/postgres:12-postgis-ram || true
-docker pull ruby:2.7.4-alpine || true
-docker pull docker.elastic.co/elasticsearch/elasticsearch:7.9.3 || true
-docker pull circleci/redis:6-alpine || true
+function docker_pull_with_retry() {
+  docker_image=$1
+
+  for i in {1..5}; do
+    docker pull "$docker_image" && break || sleep 5
```
Comment (lavaturtle) [inline packer/linux/conf/bin/bk-install-elastic-stack.sh:246]: I assume the order of operations makes this equivalent to `(docker pull "$docker_image" && break) || sleep 5`, not `docker pull "$docker_image" && (break || sleep 5)`?

Comment (lavaturtle) [reply packer/linux/conf/bin/bk-install-elastic-stack.sh:246]: I looked this up and the answer is yes: `&&` and `||` have the same precedence, so the leftmost one is carried out first. (h/t https://unix.stackexchange.com/a/88851/150331)

File: packer/linux/conf/bin/bk-install-elastic-stack.sh [hunk 2]
```diff
@@ -239,8 +239,17 @@ systemctl start "buildkite-agent"
...
+  done
+
+  true
```
Comment (lavaturtle) [inline packer/linux/conf/bin/bk-install-elastic-stack.sh]: What does this do?

**Review by lavaturtle — APPROVED** [review]
:+1:

File: packer/linux/conf/bin/bk-install-elastic-stack.sh (hunk 2 above)
Comment (anero) [reply packer/linux/conf/bin/bk-install-elastic-stack.sh]: I don't think we need this. It's a left over from a previous implementation.

**Pull request merged**